site/source/cookbook/widgets/trigger-rerender.md

```diff
@@ -0,0 +1,35 @@
+---
+layout: docs
+category: widgets
+title: Force a widget to re-render itself
+overview: Use the invalidate method to force a widget to re-render itself
+---
+
+## Objective
+
+This recipe demonstrates how a widget can force itself to re-render when it is clicked on.
+
+## Code example
+
+This code example does the following:
+
+1. Defines a `HelloWorld` widget
+2. Invokes the method `this.invalidate()` within a click handler
+
+```js
+class HelloWorld extends WidgetBase {
+    render() {
+        return v('div', {
+            onclick: this.invalidate
```

Comment:
I wonder if it's obvious to people how context is preserved here? Probably fine, just a stray thought

Reply:
Oh how do you mean?

Reply:
I just mean it's somewhat magical that you can point to this.invalidate for an onclick handler and that the context is preserved rather than in the context of the window object. But perhaps that's a stray thought and doesn't matter for the cookbook.

Reply:
It's quite nice that Dojo is doing the binding automatically! But hopefully it "just works" like people intuitively expect and if needed we can explain in a tutorial 👍